cmd/xl-storage.go

```diff
@@ -217,9 +217,35 @@ func newXLStorage(ep Endpoint) (*xlStorage, error) {
 	if env.Get("MINIO_CI_CD", "") != "" {
 		rootDisk = true
 	} else {
-		rootDisk, err = disk.IsRootDisk(path, "/")
-		if err != nil {
-			return nil, err
+		if IsDocker() || IsKubernetes() {
+			// Start with overlay "/" to check if
+			// possible the path has device id as
+			// "overlay" that would mean the path
+			// is emphemeral and we should treat it
+			// as root disk from the baremetal
+			// terminology.
+			rootDisk, err = disk.IsRootDisk(path, "/")
+			if err != nil {
+				return nil, err
+			}
+			if !rootDisk {
```

Comment:
I am not able to understand why this block is needed, if this is not root disk, aren't we happy enough ? why we need to check (something else) again ?

Reply:
because `/` in the container is not the real `/`  on the host system @vadmeste - so if you start a container with `/mnt/disk1`  then it would be referenced at `/`  - but if you did `/data/disk1` then its referenced at `/data` 

We need to look at both to know to be specifically what is the real root disk inside containers, this is an unfortunate problem that we need to have this kind of code for containers.